spring-kafka/src/main/java/org/springframework/kafka/requestreply/ReplyingKafkaTemplate.java

```diff
@@ -502,39 +506,50 @@ private static <K, V> CorrelationKey defaultCorrelationIdStrategy(
 	@Override
 	public void onMessage(List<ConsumerRecord<K, R>> data) {
 		data.forEach(record -> {
-			Header correlationHeader = record.headers().lastHeader(this.correlationHeaderName);
-			Object correlationId = null;
-			if (correlationHeader != null) {
-				correlationId = this.binaryCorrelation
-						? new CorrelationKey(correlationHeader.value())
-						: new String(correlationHeader.value(), StandardCharsets.UTF_8);
-			}
-			if (correlationId == null) {
-				this.logger.error(() -> "No correlationId found in reply: " + KafkaUtils.format(record)
-						+ " - to use request/reply semantics, the responding server must return the correlation id "
-						+ " in the '" + this.correlationHeaderName + "' header");
+			ContainerProperties containerProperties = this.replyContainer.getContainerProperties();
+			Observation observation = KafkaListenerObservation.LISTENER_OBSERVATION.observation(
```

Comment:
Do we really need to do this manually here?
Why enabling observation on the provided `GenericMessageListenerContainer` is not enough?
See more info in docs: https://docs.spring.io/spring-kafka/reference/kafka/micrometer.html#observation.

With that it feels like nothing to do in the code, unless we would like to improve docs for `ReplyingKafkaTemplate` mentioning such an observation option in the `ContainerProperties`: https://docs.spring.io/spring-kafka/reference/kafka/container-props.html#observationEnabled

Reply:
Observation is only supported on single record listeners. Batch listeners do not currently support observation. This would require another user contract to enable creating/building one observation per batch record (a batch may contain records associated to multiple observations). 
Since the ReplyingKafkaTemplate is consuming the reply through a batch listener, there's no way to automate it. I wanted to have a utility class but, in the end, the code wasn't smaller.
Did I missed something?